consensus/polybft/state_sync_manager.go

```diff
@@ -29,12 +28,41 @@ const (
 	minCommitmentSize = 2
 )
 
+type StateSyncProof struct {
+	Proof     []types.Hash
+	StateSync *contractsapi.StateSyncedEvent
+}
+
+// EncodeAbi contains logic for encoding given ABI data
+func (ssp *StateSyncProof) EncodeAbi() ([]byte, error) {
```

Comment:
Why does this data structure need to `EncodeAbi`? My logic is that anything that requires abi encoding should be inside contractsapi because its a contract call.

Reply:
Might be some legacy. I'm still working on this, so I will check if we can remove this.